File: TP1.cs
using System;
using System.Text;
// Autor: Sandra Roxana Statuto
// Se pueden agregar más frases al arreglo frases_criollo 
class Program
{
    static void Main()
    {
        string[] frasesCriollo = {
            "Hola, amigo",
            "Esto es una prueba",
            "Vocal",
            "Perros",
            "Nave"
        };

        string[] frasesTraducidas = new string[frasesCriollo.Length];

        for (int i = 0; i < frasesCriollo.Length; i++)
        {
            frasesTraducidas[i] = Traducir(frasesCriollo[i]);
        }

        Console.WriteLine("Texto en español criollo y su traducción a castellano profundo:");
        for (int i = 0; i < frasesCriollo.Length; i++)
        {
            Console.WriteLine($"{frasesCriollo[i]} -> {frasesTraducidas[i]}");
        }
    }

    static string Traducir(string palabra)
    {
        palabra = palabra.ToLower();
        StringBuilder traduccion = new StringBuilder();
        bool primeraVocalDuplicada = false;

        for (int i = 0; i < palabra.Length; i++)
        {
            char letra = palabra[i];
            if (EsVocal(letra))
            {
                if (!primeraVocalDuplicada || (i > 0 && !EsVocal(palabra[i - 1])))
                {
                    traduccion.Append(letra);
                    traduccion.Append(letra);
                    primeraVocalDuplicada = true;
                }
                else
                {
                    traduccion.Append(letra);
                }
            }
            else
            {
                traduccion.Append(letra);
            }
        }

        if (traduccion.Length > 6)
        {
            traduccion.Insert(0, "an");
        }

        char ultimaLetra = traduccion[traduccion.Length - 1];
        if (ultimaLetra == 'n' || ultimaLetra == 's' || EsVocal(ultimaLetra))
        {
            traduccion.Append(ultimaLetra == 'o' ? "so" : "sten");
        }

        return traduccion.ToString();
    }

    static bool EsVocal(char letra)
    {
        return "aeiou".Contains(letra);
    }
}

File: TP1.py
def traducir(palabra):
    # Convertir todo a minúsculas
    palabra = palabra.lower()
    traduccion = []
    primera_vocal_duplicada = False

    for letra in palabra:
        if es_vocal(letra):
            if not primera_vocal_duplicada or (len(traduccion) > 0 and not es_vocal(traduccion[-1])):
                traduccion.append(letra * 2)
                primera_vocal_duplicada = True
            else:
                traduccion.append(letra)
        else:
            traduccion.append(letra)

    if len(traduccion) > 6:
        traduccion.insert(0, "an")

    ultima_letra = traduccion[-1]
    if ultima_letra in ['n', 's'] or es_vocal(ultima_letra):
        traduccion.append("so" if ultima_letra == 'o' else "sten")

    return ''.join(traduccion)

def es_vocal(letra):
    return letra in 'aeiou'

frases_criollo = [
    "Hola, amigo",
    "Esto es una prueba",
    "Vocal",
    "Perros",
    "Nave"
]

frases_traducidas = []

for frase in frases_criollo:
    frases_traducidas.append(traducir(frase))

print("Texto en español criollo y su traducción a castellano profundo:")
for i in range(len(frases_criollo)):
    print(f"{frases_criollo[i]} -> {frases_traducidas[i]}")
